### PR TITLE
Add leave page protection

### PR DESCRIPTION
**Pull request by loneil**
# Description
Typical browser warning when a user closes an in-progress page

Designer
Turn on when they go to step 2
Turn off when it succeeds

Viewer
Turn on when loaded
Turn off when succeeds

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-wxpbtr-tools.pathfinder.gov.bc.ca/dashboard?id=common-hosted-form-service-pr-40

File: app/frontend/src/components/designer/FormDesigner.vue [hunk 1]
```diff
@@ -382,6 +399,17 @@ export default {
         this.getFormSchema();
       }
     },
+    beforeDestroy() {
+      window.onbeforeunload = null;
+    }
+  },
+  watch: {
+    designerStep(newValue) {
```
Comment (parc-jason) [inline app/frontend/src/components/designer/FormDesigner.vue:404]: On Safari and Chrome, this isn't kicking in for me.
Going to the submitter side, I get the browser warning when leaving, but can't get it going in designer.

File: app/frontend/src/components/designer/FormDesigner.vue [hunk 2]
```diff
@@ -382,6 +399,17 @@ export default {
...
+      if (newValue === 2) {
```
Comment (parc-jason) [inline app/frontend/src/components/designer/FormDesigner.vue:405]: wait, it does kick in, but only when i close the tab.  not leave the screen (ie navigate around in the app itself).

**Review by jujaga — APPROVED** [review]

File: app/frontend/src/components/designer/FormDesigner.vue (hunk 1 above)
Comment (jujaga) [reply app/frontend/src/components/designer/FormDesigner.vue:404]: This is a vue-router navigation issue. Will be looked at in future PR.

File: app/frontend/src/components/designer/FormDesigner.vue (hunk 2 above)
Comment (jujaga) [reply app/frontend/src/components/designer/FormDesigner.vue:405]: This is a vue-router navigation issue. Will be looked at in future PR.

**Pull request merged**